NServiceBus.FileBasedRouting/PublishRoutingConnector.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            var eventType = context.Message.MessageType;
+            var routingStrategies = Route(eventType).ToList();
+            if (routingStrategies.Count == 0)
+            {
+                //No subscribers for this message.
+                return;
+            }
+
+            context.Headers[Headers.MessageIntent] = MessageIntentEnum.Publish.ToString();
+
+            try
+            {
+                await stage(this.CreateOutgoingLogicalMessageContext(context.Message, routingStrategies, context)).ConfigureAwait(false);
+            }
+            catch (QueueNotFoundException ex)
+            {
+                throw new Exception($"The destination queue '{ex.Queue}' could not be found. The destination may be misconfigured for this kind of message ({eventType}) in the MessageEndpointMappings of the UnicastBusConfig section in the configuration file. It may also be the case that the given queue hasn\'t been created yet, or has been deleted.", ex);
+            }
+        }
+
+        IEnumerable<UnicastRoutingStrategy> Route(Type messageType)
+        {
+            var routes = routingTable.GetRoutesFor(messageType);
+            var selectedDestinations = SelectDestinationsForEachEndpoint(routes);
+            return selectedDestinations.Select(destination => new UnicastRoutingStrategy(destination));
+        }
+
+        HashSet<string> SelectDestinationsForEachEndpoint(IEnumerable<UnicastRouteGroup> routeGroups)
+        {
+            //Make sure we are sending only one to each transport destination. Might happen when there are multiple routing information sources.
+            var addresses = new HashSet<string>();
+
+            foreach (var group in routeGroups)
+            {
+                if (group.EndpointName == null) //Routing targets that do not specify endpoint name
+                {
+                    //Send a message to each target as we have no idea which endpoint they represent
+                    foreach (var subscriber in group.Routes)
+                    {
+                        foreach (var address in ResolveRoute(subscriber))
+                        {
+                            addresses.Add(address);
+                        }
+                    }
+                }
+                else
+                {
+                    var candidates = group.Routes.SelectMany(ResolveRoute).ToArray();
+                    var selected = distributionPolicy.GetDistributionStrategy(group.EndpointName, DistributionStrategyScope.Publish).SelectReceiver(candidates);
+                    addresses.Add(selected);
+                }
+            }
+
+            return addresses;
+        }
+
+        IEnumerable<string> ResolveRoute(UnicastRoute route)
+        {
+            if (route.Instance != null)
+            {
+                yield return resolveTransportAddress(route.Instance);
+            }
+            else if (route.PhysicalAddress != null)
```

Comment:
Same as above, if we continue to use `UnicastRoute` then I think we should account for it. That said, we can replace it with a specialized type that does not allow for that big variety of route specifications.